src/Meshes/abstractMesh.ts

```diff
@@ -1713,6 +1745,64 @@ export class AbstractMesh extends TransformNode implements IDisposable, ICullabl
         return pickingInfo;
     }
 
+    /**
+     * Picks this mesh with a sphere
+     * @param sphere picking sphere in world coordinates
+     * @param skipBoundingInfo a boolean indicating if we should skip the bounding info check
+     * @returns the picking info
+     */
+    public pickWithSphere(sphere: BoundingSphere, skipBoundingInfo = false): PickingInfo {
```

Comment:
We are trying to avoid adding new elements to mesh/abstract mesh.

Is there a reason to include the pickWithSphere here? what is the (abstract) use of this outside of XR near interactions?

Reply:
I put it there because this function's logic is very similar to the above one, `intersects`. Instead of intersecting with a ray, we intersect with a sphere. So it felt like it belonged here. 
If we really want to avoid adding new members to `abstractMesh`, where do you suggest to put it ?  `WebXRNearInteraction` ?

Reply:
Yes, I would keep it there for now. 
abstractMesh has `intersectsWithMesh` as well, but this is involving picking, which should technically be on the scene level.

Reply:
So I think the best is to keep the function here, and change the name to `intersectsSphere`, which is more accurate and consistent with the above function

Reply:
intersects returns a boolean. and has little to do with picking info.

I totally understand the need to put it somewhere else, but this is a very specialized function that is only being used by the near interaction feature. The fact is that we didn't need it until now.
We are trying to minimize the impact on large classes like the mesh and the scene by not adding too much "specialized" functionality to it.